frontend/src/components/Playback/PlayerElement.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <template v-if="mediaElement">
+    <Teleport :to="teleportTarget">
+      <component
+        :is="mediaElement"
+        ref="mediaElementRef"
+        :poster="posterUrl"
+        autoplay
+        crossorigin="anonymous"
+        playsinline
+        :loop="isLoopingOnce" />
+    </Teleport>
+  </template>
+</template>
+
+<script setup lang="ts">
+/**
+ * Shaka Player needs to be unmounted for it to be properly disposed. However,
+ * since this component will always be mounted in the app (as it's inside the top-level App component),
+ * it doesn't make sense to add code for handling the unmount process.
+ */
+// @ts-expect-error - This module doesn't have typings
+import muxjs from 'mux.js';
+// @ts-expect-error - This module doesn't have typings
+import shakaPlayer from 'shaka-player/dist/shaka-player.compiled';
+import { computed, watch } from 'vue';
+import { isNil } from 'lodash-es';
+import { playbackManagerStore, RepeatMode } from '@/store';
+import { mediaElementRef } from '@/store/playbackManager';
+import { getImageInfo } from '@/utils/images';
+
+const playbackManager = playbackManagerStore();
+let isShakaAttached = false;
+const shaka = new shakaPlayer.Player();
+
+window.muxjs = muxjs;
+window.player = shaka;
+shakaPlayer.polyfill.installAll();
+
+const mediaElement = computed<'audio' | 'video' | undefined>(() => {
+  if (playbackManager.currentlyPlayingMediaType === 'Audio') {
+    return 'audio';
+  } else if (playbackManager.currentlyPlayingMediaType === 'Video') {
+    return 'video';
+  }
+});
+
+/**
+ * If the player is a video element and we're
```

Comment:
And we're what? :p

Reply:
```
/**
 * If the player is a video element and we're
 */
const teleportTarget = computed<'body' | '.video-container'>(() => {
  return mediaElement.value === 'audio' ? 'body' : '.video-container';
});
```

if that can help you :)

Reply:
@ThibaultNocchi IIRC I wanted to continue with something like this:

``If the player is a video element and we're in the PiP player or fullscreen video playback, we need to ensure the DOM elements are mounted before the teleport target is ready``